app/models/cart.rb

```diff
@@ -1,3 +1,29 @@
 class Cart < ApplicationRecord
   has_many :line_items, dependent: :destroy
+
+  def add_product(product)
+    current_item = line_items.find_by(product_id: product.id)
+    if current_item
+      current_item.quantity += 1
+    else
+      current_item = line_items.build(product_id: product.id)      
+    end
+    current_item
+  end
+
+  # get total number of items in the cart or 0
+  def total_num_items
+    line_items.reduce(0) { |sum, item| sum + item.quantity }     
```

Comment:
A more succinct Ruby way of writing this would be:

```suggestion
    line_items.sum(&:quantity)
```